cli/src/config.rs

```diff
@@ -268,62 +257,84 @@ impl ConfigEnv {
         }
     }
 
-    fn config_path(self) -> Result<ConfigPath, ConfigError> {
+    // Returns a config path, potentially creating a new one if told. Can return
+    // None if no config exists and we're not creating a new one.
+    // Also bubbles up errors if encountered.
+    fn new_or_existing_config_path(
+        self,
+        handle_missing: HandleMissingConfig,
+    ) -> Result<Option<PathBuf>, ConfigError> {
         if let Some(path) = self.jj_config {
             // TODO: We should probably support colon-separated (std::env::split_paths)
-            return Ok(ConfigPath::new(Some(PathBuf::from(path))));
+            let path = PathBuf::from(path);
+            if handle_missing == HandleMissingConfig::Create && !path.exists() {
+                create_config_file(&path)?;
+            }
```

Comment:
Was this required because of some other changes? Or does this represent a change in behavior?

Reply:
I combined the two functions into one, with an argument to create the file when needed. This is existing behavior.

Reply:
Perhaps, `create_config_file()` can be moved to caller? It's generally easier to test if the function had no side effect.